src/python/dxpy/bindings/__init__.py

```diff
@@ -657,6 +657,33 @@ def _wait_on_close(self, timeout=3600*24*1, **kwargs):
             i += 1
             elapsed += wait
 
+    def _wait_until_parts_uploaded(self, timeout=60, **kwargs):
+        elapsed = 0
+        i = 0
+        while True:
+            parts = self.describe(fields={'parts'}, **kwargs)["parts"]
+            state = self._get_state(**kwargs)
+            if state == "closed":
+                # parts of closed files must have been uploaded successfully
+                break
+            if not parts:
+                break
+            else:
+                is_uploaded = True
+                for key in parts:
+                    if parts[key].get(state, 'complete') != 'complete':
+                        is_uploaded = False
+                        break
+                if is_uploaded:
+                    break
+            if elapsed >= timeout or elapsed < 0:
```

Comment:
Consider the case where is_uploaded is true but we happened to exceed the timeout

Reply:
Got it. Thanks!